src/main/java/com/gluonhq/NativeBaseMojo.java

```diff
@@ -136,43 +134,44 @@
 
     private ProcessDestroyer processDestroyer;
 
-    Configuration clientConfig;
+    ProjectConfiguration clientConfig;
 
     public void execute() throws MojoExecutionException {
-        configOmega();
+        if (!getGraalHome().isPresent()) {
+            throw new MojoExecutionException("Graal installation directory not found." +
+                    " Either set GRAAL_HOME as an environment variable or" +
+                    " set graalHome in client-plugin configuration");
+        }
+        configSubstrate();
     }
 
-    private void configOmega() {
-        clientConfig = new Configuration();
-        clientConfig.setGraalLibsVersion(graalLibsVersion);
+    private void configSubstrate() {
+        clientConfig = new ProjectConfiguration();
+        clientConfig.setGraalPath(getGraalHome().get());
         clientConfig.setJavaStaticSdkVersion(javaStaticSdkVersion);
         clientConfig.setJavafxStaticSdkVersion(javafxStaticSdkVersion);
 
         String osname = System.getProperty("os.name", "Mac OS X").toLowerCase(Locale.ROOT);
-        TargetTriplet hostTriplet;
+        Triplet hostTriplet;
         if (osname.contains("mac")) {
-            hostTriplet = new TargetTriplet(Constants.AMD64_ARCH, Constants.HOST_MAC, Constants.TARGET_MAC);
+            hostTriplet = new Triplet(Constants.ARCH_AMD64, Constants.VENDOR_APPLE, Constants.OS_DARWIN);
         } else if (osname.contains("nux")) {
-            hostTriplet = new TargetTriplet(Constants.AMD64_ARCH, Constants.HOST_LINUX, Constants.TARGET_LINUX);
+            hostTriplet = new Triplet(Constants.ARCH_AMD64, Constants.VENDOR_LINUX, Constants.OS_LINUX);
         } else {
             throw new RuntimeException("OS " + osname + " not supported");
         }
-        clientConfig.setHost(hostTriplet);
+        clientConfig.setHostTriplet(hostTriplet);
```

Comment:
There is no need to set the host triplet, it will be set in Substrate by default with `Triplet.fromCurrentOS()`.